src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -69,5 +70,36 @@ private static string EnsureTrailingCharacter(string path, char trailingCharacte
 
             return path + trailingCharacter;
         }
+
+        private static bool CheckIfFileSystemIsCaseInsensitive()
+        {
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                return true;
+            }
+            else if (RuntimeEnvironmentHelper.IsLinux)
+            {
+                return false;
+            }
+            else
+            {
+                var curDir = Directory.GetCurrentDirectory();
+                if (curDir.Length > 1)
+                {
+                    return Directory.Exists(curDir.ToUpper()) && Directory.Exists(curDir.ToLower());
```

Comment:
The user could possibly be in a folder that also has the lowercase spelling next to it. Edge case, but still.

How about:
1. Get NuGetFolderPath.NuGetHome and NuGetFolderPath.Temp
1. For both of these directories keep moving to the parent directory until you find one that actually exists
1. Verify that for both folders both the upper and lowercase paths exist for the two directories you end up with

a little more complicated, but I think this would give a much better result.